src/main/java/com/google/api/codegen/metacode/InitCodeGenerator.java

```diff
@@ -0,0 +1,167 @@
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.lang3.NotImplementedException;
+
+/**
+ * InitCodeGenerator generates an InitCode object for a given method and
+ * initialization field structure (as constructed by FieldStructureParser).
+ */
+public class InitCodeGenerator {
+  private Set<String> symbolTable = new HashSet<>();
+  private List<InitCodeLine> initLineSpecs = new ArrayList<>();
+
+  /**
+   * Generates the InitCode for a method, where the input of the function
+   * representing the method will take a request object.
+   */
+  public InitCode generateRequestObjectInitCode(Method method,
+      Map<String, Object> initFieldStructure) {
+    InitCodeLine lastLine = generateSampleCodeInit("request", method.getInputType(),
+        initFieldStructure);
+    initLineSpecs.add(lastLine);
+    FieldSetting requestField = FieldSetting.create(method.getInputType(), "request",
+        lastLine.getIdentifier(), lastLine.getInitValueConfig());
+    List<FieldSetting> outputFields = Arrays.asList(requestField);
+    return InitCode.create(initLineSpecs, outputFields);
+  }
+
+  /**
+   * Generates the InitCode for a method, where the input of the function
+   * representing the method will take the given set of fields.
+   */
+  public InitCode generateRequestFieldInitCode(Method method,
+      Map<String, Object> initFieldStructure, Iterable<Field> fields) {
+
+    Map<String, Object> filteredInit = new HashMap<>();
+    for (Field field : fields) {
+      Object subStructure = initFieldStructure.get(field.getSimpleName());
+      if (subStructure != null) {
+        filteredInit.put(field.getSimpleName(), subStructure);
+      } else {
+        filteredInit.put(field.getSimpleName(), InitValueConfig.create());
+      }
+    }
+
+    InitCodeLine lastLine = generateSampleCodeInit("request", method.getInputType(), filteredInit);
+    if (!(lastLine instanceof StructureInitCodeLine)) {
+      throw new IllegalArgumentException(
+          "Expected method request to be a message, found " + lastLine.getClass().getName());
+    }
+    StructureInitCodeLine requestInitCodeLine = (StructureInitCodeLine) lastLine;
+    return InitCode.create(initLineSpecs, requestInitCodeLine.getFieldSettings());
+  }
+
+  private String getNewSymbol(String desiredName) {
+    String actualName = desiredName;
+    int i = 2;
+    while (symbolTable.contains(actualName)) {
+      actualName = desiredName + i;
+    }
+    symbolTable.add(actualName);
+    return actualName;
+  }
+
+  private InitCodeLine generateSampleCodeInit(String suggestedName, TypeRef typeRef,
+      Object initFieldStructure) {
+    if (initFieldStructure instanceof InitValueConfig) {
+      InitValueConfig initValueConfig = (InitValueConfig) initFieldStructure;
+      String identifier = getNewSymbol(suggestedName);
+      return SimpleInitCodeLine.create(typeRef, identifier, initValueConfig);
+    }
+
+    if (typeRef.isMessage() && !typeRef.isRepeated()) {
```

Comment:
I was trying to follow the same pattern everywhere: Map, List, simple type. (e.g. see FieldStructureParser.) I know that the condition checks might be slightly more optimized by putting it after, but I think the consistency of ordering is higher value.